lib/memo_wise.rb

```diff
@@ -3,6 +3,46 @@
 require "memo_wise/version"
 
 module MemoWise
-  class Error < StandardError; end
-  # Your code goes here...
+  def initialize(*values)
+    @_memo_wise = {}
+    super
+  end
+
+  def self.prepended(target)
+    class << target
+      # Implements memoization for the given method name.
+      #
+      # @param method_name [Symbol]
+      #   Name of method for which to implement memoization.
+      def memo_wise(method_name)
+        not_memoized_name = "#{method_name}_not_memoized".to_sym
+        alias_method not_memoized_name, method_name
+        private not_memoized_name
+
+        module_eval <<-END_OF_METHOD, __FILE__, __LINE__ + 1
+          def #{method_name}(*args)
+            key = ["#{method_name}", args].freeze
```

Comment:
Why convert `method_name` to String here? I am very confident it will be measurably faster for this key lookup comparison to be a tuple of (Symbol, Array) than it would be for (String, Array)?

Reply:
Switched it to be a symbol here, but it needs to be interpolated because it's within the larger `module_eval` string